cosign/README.md

```diff
@@ -0,0 +1,125 @@
+# Cosign Verifier
+
+## Introduction
+
+In order to provide signatures over arbitrary artifacts, Sigstore performs three major operations[1]:
+
+1. OIDC Issuance, vouch that a client is in control of an identity
+2. Associate short-lived public key certificates with these identities (from a Certificate Authority), and publish these certificates to an Identity Log.
+3. Publish a long-lived signature over an artifact (or artifact meta-data) to an Artifact Log, allowing verifiers to check its validity.
+
+The first and second operations are performed by a system called Fulcio, which functions as a Certificate Authority and transparency log for a namespace of OIDC identities (Identity Log).
+The third operation is performed by a system called Rekor, a transparency log for artifact signatures (Artifact Log).
+And cosign is the reference Sigstore client implementation to sign arbitrary artifacts. With these components, Sigstore completed trust setup, signing and verification[2].
+
+```mermaid
+flowchart TB
+  %% OIDC Flow
+  Signer["💻 Signer"] -->|① OIDC Challenge Start| OIDC["@ OIDC Provider"]
+  OIDC -->|② OIDC Response| Signer
+  
+  %% Certificate Issuance via Fulcio
+  Signer -->|③ Issuance Request| Fulcio["📜 Fulcio (Certificate Authority)"]
+  Fulcio -->|④ Signed Certificate| Signer
+  Fulcio -->|Publish to Identity Log| IdentityLog
+  
+  %% Signing & Publishing Artifacts
+  Signer -->|⑤ Sign and Publish Artifact| ArtifactLog
+  Signer -->|⑤ Sign and Publish Artifact| ArtifactRepo["📦 Artifact Repository"]
+  
+  %% Identity and Artifact Logs
+  subgraph Logs["Transparency Logs"]
+    subgraph IdentityLog["📜 Identity Log (Fulcio)"]
+      ID1["UUID: 5a88c7 SAN: user#64;github.com"]
+      ID2["UUID: 7a40cf SAN: user2#64;github.com"]
+      ID3["UUID: b76141 SAN: user#64;name.io"]
+    end
+    
+    subgraph ArtifactLog["📝 Artifact Log (Rekor)"]
+      AID1["UUID: 5a88c7 SAN: io.github.pytorch"]
```

Comment:
`Rekor` seems do not utilize `SAN`.

Reply:
As `SAN` stands subject alternative name, plus this is not in the critical path of either verifier implementation or a core concept sigstore, skip the details for now.